Common/cpp/SharedItems/Shareables.h

```diff
@@ -273,6 +274,30 @@ class ShareableHostObject : public Shareable {
   std::shared_ptr<jsi::HostObject> hostObject_;
 };
 
+class ShareableHostFunction : public Shareable {
+ public:
+  ShareableHostFunction(
+      const std::shared_ptr<JSRuntimeHelper> &runtimeHelper,
+      jsi::Runtime &rt,
+      jsi::Function function)
+      : Shareable(HostFunctionType),
+        hostFunction_(
+            (assert(function.isHostFunction(rt)),
+             function.getHostFunction(rt))),
+        name_(function.getProperty(rt, "name").asString(rt).utf8(rt)),
+        paramCount_(function.getProperty(rt, "length").asNumber()) {}
+
+  jsi::Value toJSValue(jsi::Runtime &rt) override {
+    return jsi::Function::createFromHostFunction(
+        rt, jsi::PropNameID::forUtf8(rt, name_), paramCount_, hostFunction_);
+  }
```

Comment:
I'm not sure what the overhead from `createFromHostFunction` is, but we should probably look into caching the results from `toJSValue`.

Reply:
I guess this might be already covered by the shareable cache:
https://github.com/software-mansion/react-native-reanimated/blob/a38678d6ada894f4c42461de806a90d2a94a688a/src/reanimated2/shareables.ts#L49-L54